src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -215,20 +215,21 @@ public static String formatTime12H(Date date) {
         return sdf.format(date);
     }
 
-    public static String formatDateTimeForComments(Date date) {
+    public static String formatDateTimeForComments(Date date, double sessionTimeZone) {
         if (date == null) {
             return "";
         }
         SimpleDateFormat sdf = null;
         Calendar c = Calendar.getInstance(SystemParams.TIME_ZONE);
         c.setTime(date);
+        String timeZone = getCustomIdOfTimeZone(sessionTimeZone);
         if (c.get(Calendar.HOUR_OF_DAY) == 12 && c.get(Calendar.MINUTE) == 0) {
```

Comment:
`Calendar.HOUR_OF_DAY` stores time in 24-hour format. If it is 12 means it is `12 NOON` only. 
Apologies, If I misunderstood anything. Can you please elaborate a bit more.